### PR TITLE
[fuzz] Refactored Health Checker Impl Tests

### PR DESCRIPTION
**Pull request by zasweq**
Signed-off-by: Zach <zasweq@google.com>

Commit Message: Refactored Health Checker Impl Tests
Additional Description: Refactored Health Checker Impl Tests for use in fuzzing
Risk Level:
Testing:
Docs Changes:
Release Notes:

### CODE REVIEW

File: test/common/upstream/health_checker_impl_test.cc [hunk 1]
```diff
@@ -2439,7 +2885,7 @@ TEST(TcpHealthCheckMatcher, match) {
   EXPECT_TRUE(TcpHealthCheckMatcher::match(segments, buffer));
 }
 
-class TcpHealthCheckerImplTest : public testing::Test, public HealthCheckerTestBase {
+class TcpHealthCheckerImplTest : public testing::Test, public TcpHealthCheckerImplTestBase {
```
Comment (asraa) [inline test/common/upstream/health_checker_impl_test.cc:2888]: are the mock timers in this class needed in the base? if you're not sure yet -- maybe scope this down to HTTP and leave this for the TCP PR?

**Review by asraa — COMMENTED** [review]
Thanks! Other than the comment lgtm

File: test/common/upstream/health_checker_impl_test.cc (hunk 1 above)
Comment (zasweq) [reply test/common/upstream/health_checker_impl_test.cc:2888]: Yup - let me scope it down to http.

**Review by asraa — APPROVED** [review]
Thanks!

**Pull request merged**